forui/lib/src/widgets/popover_menu.dart

```diff
@@ -0,0 +1,286 @@
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty('controller', controller))
+      ..add(DiagnosticsProperty('style', style))
+      ..add(EnumProperty('divider', divider))
+      ..add(DiagnosticsProperty('menuAnchor', menuAnchor))
+      ..add(DiagnosticsProperty('childAnchor', childAnchor))
+      ..add(DiagnosticsProperty('shift', shift))
+      ..add(FlagProperty('hideOnTapOutside', value: hideOnTapOutside, ifTrue: 'hideOnTapOutside'))
+      ..add(
+        FlagProperty(
+          'ignoreDirectionalPadding',
+          value: ignoreDirectionalPadding,
+          ifTrue: 'ignoreDirectionalPadding',
+        ),
+      )
+      ..add(StringProperty('semanticLabel', semanticLabel))
+      ..add(FlagProperty('autofocus', value: autofocus, ifTrue: 'autofocus'))
+      ..add(DiagnosticsProperty('focusNode', focusNode))
+      ..add(ObjectFlagProperty.has('onFocusChange', onFocusChange));
+  }
+}
+
+class _FPopoverMenuState extends State<FPopoverMenu> with SingleTickerProviderStateMixin {
+  late FPopoverController _controller;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = widget.controller ?? FPopoverController(vsync: this);
+  }
+
+  @override
+  void didUpdateWidget(covariant FPopoverMenu old) {
+    super.didUpdateWidget(old);
+    if (widget.controller == old.controller) {
+      return;
+    }
+
+    if (old.controller != null) {
+      _controller.dispose();
+    }
+    _controller = widget.controller ?? FPopoverController(vsync: this);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final style = widget.style ?? context.theme.popoverMenuStyle;
+    return (widget._tappable ? FPopover.tappable : FPopover.new)(
+      controller: _controller,
+      style: style,
+      followerAnchor: widget.menuAnchor,
+      targetAnchor: widget.childAnchor,
+      shift: widget.shift,
+      hideOnTapOutside: widget.hideOnTapOutside,
+      ignoreDirectionalPadding: widget.ignoreDirectionalPadding,
+      autofocus: widget.autofocus,
+      focusNode: widget.focusNode,
+      onFocusChange: widget.onFocusChange,
+      followerBuilder: (context, _, __) => ConstrainedBox(
+        constraints: BoxConstraints(maxWidth: style.maxWidth),
+        child: FTileGroup.merge(
+          semanticLabel: widget.semanticLabel,
+          style: style.tileGroupStyle,
+          divider: widget.divider,
+          children: widget.menu,
+        ),
+      ),
+      target: widget.child,
+    );
+  }
+
+  @override
+  void dispose() {
+    if (widget.controller == null) {
+      _controller.dispose();
+    }
+    super.dispose();
+  }
+}
+
+/// A [FPopoverMenuStyle]'s style.
+final class FPopoverMenuStyle extends FPopoverStyle {
+  /// The popover's default shadow in [FPopoverMenuStyle.inherit].
+  static const shadow = [
+    BoxShadow(
+      color: Color(0x0d000000),
+      offset: Offset(0, 1),
+      blurRadius: 2,
+    ),
```

Comment:
I noticed that we add shadows throughout all of our overlay widgets. Should this be something that we inherit?

Reply:
The shadow is different from the other overlay.

Reply:
Should they be different?

Reply:
Yup, I originally used the same shadow for popovers & popover menus. However, you mentioned that the shadows for the popover menu was too heavy which resulted in the shadows being changed.

Reply:
Ah I see. If that's the case, would it make sense for both of them to adopt the lighter shadow? Imo, less is better than having too much. In addition, there's no gurantee that a popover is larger than a popover menu.

The popover menu is actually a good example that a popover can be small.

Reply:
Ehh sure, no strong opinion on it.